habitat/datasets/object_nav/object_nav_dataset.py

```diff
@@ -74,15 +109,7 @@ def from_json(
 
                 episode.scene_id = os.path.join(scenes_dir, episode.scene_id)
 
-            for i in range(len(episode.goals)):
-                episode.goals[i] = ObjectGoal(**episode.goals[i])
-
-                for vidx, view in enumerate(episode.goals[i].view_points):
-                    view_location = ObjectViewLocation(**view)
-                    view_location.agent_state = AgentState(
-                        **view_location.agent_state
-                    )
-                    episode.goals[i].view_points[vidx] = view_location
+            episode.goals = goals_by_category[episode.goals]
```

Comment:
Maybe avoid using `goals` field and re-create key to avoid confusions:
```suggestion
            episode.goals = goals_by_category[goals_key(episode)]
```